crates/wasi-common/src/snapshots/wasi_snapshot_preview1.rs

```diff
@@ -361,20 +361,21 @@ impl<'a> WasiSnapshotPreview1 for WasiCtx {
             buf.push(io::IoSliceMut::new(slice));
         }
 
-        let mut entry = self.get_entry_mut(fd)?;
+        let entry = self.get_entry(fd)?;
         let file = entry
-            .as_descriptor_mut(
+            .as_descriptor(
                 types::Rights::FD_READ | types::Rights::FD_SEEK,
                 types::Rights::empty(),
             )?
-            .as_file_mut()?;
+            .as_file()?;
 
         if offset > i64::max_value() as u64 {
             return Err(Errno::Io);
         }
 
         let host_nread = match file {
             Descriptor::OsHandle(fd) => {
+                let mut fd: &File = fd;
```

Comment:
What is the `mut` here for?

Reply:
The way this technically works is it's dispatching to the `impl Seek for &File` where the seek method takes `&mut self`. This means that we're calling seek on `&mut &File`, so the `mut` here is needed to create the `&mut &File` binding.

Reply:
This is to be able to mutate the reference to reference to `File`. Another way of expressing it would be

```rust
(&mut (&*fd as &File)).read_vectored(...)
```

This looks too cluttered for me, hence why I’ve decided to `let`-bind it first. However, if you know of a better way of handling this, lemme know!

Reply:
@alexcrichton again beat me to it! Seconds, seconds!